source/common/tls/default_tls_certificate_selector.cc

```diff
@@ -182,14 +184,35 @@ DefaultTlsCertificateSelector::findTlsContext(absl::string_view sni, bool client
       // The selected ctx must adhere to OCSP policy
       return false;
     }
+    // if the client is ECDSA-capable and the context is ECDSA, we check if it is capable of handling 
+    // the curves in the cert in a given TlsContext
+    if (client_ecdsa_capabilities.has_value() && ctx.is_ecdsa_) {
```

Comment:
I think it would be simpler to replace `bool is_ecdsa_` in the context with the value for it's cert curve (and some sentinel value for rsa/non-EC). That'll move some of this code from the handshake path to the config load path.